File: src/main/java/hacker/solution/SubArray.java
package hacker.solution;

import java.util.Scanner;

public class SubArray {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = scan.nextInt();
        }
        scan.close();
        int negCount = 0;
        for (int i=0;i<a.length;i++) {
            for (int j=0;j<a.length;j++) {
                if (sumSub(a,i,j)<0) negCount++;
            }
        }
        System.out.println(negCount);
    }
    private static int sumSub(int[] arr,int i, int j) {
        int sum=0;
        for (int k=i;k<=j;k++) sum+=arr[k];
        return sum;
    }
}

File: src/main/kotlin/hacker/solutionk/SubArray.kt
package hacker.solutionk

import java.util.*

object SubArray {
    @JvmStatic
    fun main(args: Array<String>) {
        val scan = Scanner(System.`in`)
        val n = scan.nextInt()
        val a = IntArray(n)
        for (i in 0 until n) {
            a[i] = scan.nextInt()
        }
        scan.close()
        var negCount = 0
        for (i in a.indices) {
            for (j in a.indices) {
                if (sumSub(a, i, j) < 0) negCount++
            }
        }
        println(negCount)
    }

    private fun sumSub(arr: IntArray, i: Int, j: Int): Int {
        var sum = 0
        for (k in i..j) sum += arr[k]
        return sum
    }
}
